src/build-config.ts

```diff
@@ -93,69 +94,68 @@ function buildInputConfig(
   const plugins: Plugin[] = (
     dts
       ? [
-          ...commonPlugins,
-          useTypescript &&
-            require('rollup-plugin-dts').default({
-              compilerOptions: {
-                ...tsCompilerOptions,
-                declaration: true,
-                noEmit: false,
-                noEmitOnError: true,
-                emitDeclarationOnly: true,
-                checkJs: false,
-                declarationMap: false,
-                skipLibCheck: true,
-                preserveSymlinks: false,
-                target: 'esnext',
-                module: 'esnext',
-                incremental: false,
-                jsx: tsCompilerOptions.jsx || 'react',
-              },
-            }),
-        ]
+        ...commonPlugins,
```

Comment:
we need to check if you're using `useTypescript` here otherwise js project without ts dependency will fail